src/cascadia/QueryExtension/ILLMProvider.idl

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace Microsoft.Terminal.Query.Extension
+{
+    interface ILLMProvider
+    {
+        void ClearMessageHistory();
+        void SetSystemPrompt(String systemPrompt);
+        void SetContext(IContext context);
+
+        Windows.Foundation.IAsyncOperation<IResponse> GetResponseAsync(String userPrompt);
+    }
+
+    interface IResponse
+    {
+        String Message { get; };
+        Boolean IsError { get; };
+    };
+
+    interface IContext
+    {
+        String ActiveCommandline { get; };
```

Comment:
Is the expectation that over time, this interface would grow with different types of context that Terminal would supply to extensions?

Reply:
Yes!